src/rules/no-untyped-mock-factory.ts

```diff
@@ -0,0 +1,87 @@
+import { AST_NODE_TYPES, TSESTree } from '@typescript-eslint/utils';
+import {
+  createRule,
+  getAccessorValue,
+  isFunction,
+  isSupportedAccessor,
+  isTypeOfJestFnCall,
+} from './utils';
+
+const findModuleName = (
+  node: TSESTree.Literal | TSESTree.Node,
```

Comment:
nit: iirc `TSESTree.Literal` is a subset of `TSESTree.Node`, so you shouldn't need it here